# BITS F464 : Machine Learning Assignment 2

Adit Rastogi <br>
2022A7PS1330H <br><br>
Deeptansh Gupta <br>
2022A7PS1336H <br><br>
Anirban Nayak <br>
2022A3PS0705H <br>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.preprocessing import LabelEncoder

#html export
import plotly.io as pio
pio.renderers.default = 'notebook'

#pdf export
!pip install Pyppeteer
!pyppeteer-install

chromium is already installed.


In [2]:
#please fill the dataset path as required
data_p1 = pd.read_csv("./MIMIC/221.csv")
data_p2 = pd.read_csv("./MIMIC/230.csv")

## Data Pre-processing For Both Patients

In [3]:
print(data_p1.shape)
data_p1.head()

(24999, 7)


RESP  BP-S  BP-D  SpO2  HR  PULSE  Anomaly
0    28    81   120    98  74     73        0
1    29    81   120    98  74     73        0
2    30    81   120    98  74     73        0
3    30    81   120    99  74     73        0
4    29    81   120    99  73     73        0

In [4]:
print(data_p2.shape)
data_p2.head()

(25000, 7)


RESP  BP-S  BP-D  SpO2  HR  PULSE  Anomaly
0    15    54    73    97  72     71        0
1    15    54    73    97  72     71        0
2    15    54    73    97  72     71        0
3    15    54    73    97  72     71        0
4    15    55    74    97  72     71        0

In [5]:
df_p1 = pd.DataFrame(data_p1)
df_p2 = pd.DataFrame(data_p2)

#### Creating a Describe Function

In [6]:
def describe(df):
    describe_df = df.describe()

    # Calculating median and standard deviation
    median = df.median(numeric_only=True)
    std_dev = df.std(numeric_only=True)

    # Appendind median and standard deviation to describe_df
    describe_df.loc['median'] = median
    describe_df.loc['std'] = std_dev

    # Removing 25%, 50%, and 75% statistics
    describe_df = describe_df.drop(['25%', '50%', '75%'])

    # Reordering the rows
    describe_df = describe_df.loc[['count', 'mean', 'median', 'std', 'min', 'max']]

    # Styling
    styled_summary = (describe_df.style
                      .format("{:.2f}")  # Format numbers to 2 decimal places
                      .background_gradient(cmap='coolwarm')  # Add color gradient
                      .set_table_styles([{'selector': 'thead th', 'props': [('background-color', '#404040'), ('color', 'white')]}])  # Style headers
                      )

    display(styled_summary)
    return 

In [7]:
print("Patient 1")
describe(df_p1)
print("Patient 2")
describe(df_p2)

Patient 1


Patient 2


### Finding columns with missing values

In [18]:
def missing(df):
    missing_values_per_column = df.isnull().sum()
    columns_with_missing_values = missing_values_per_column[missing_values_per_column > 0]
    print("Columns with missing values:\n", columns_with_missing_values)
    return 

In [20]:
print("Patient 1")
missing(df_p1)
print("Patient 2")
missing(df_p2)

Patient 1
Columns with missing values:
 Series([], dtype: int64)
Patient 2
Columns with missing values:
 Series([], dtype: int64)


- There are no Missing Values in Both the DataSets

### Checking for Duplicates in the Data Frame

In [25]:
def check_dup(df):
    duplicates = df.duplicated()
    num_duplicates = duplicates.sum()

    print("Number of duplicate rows:", num_duplicates)
    if num_duplicates > 0:
        print("\nDuplicate rows:")
        print(df[duplicates])
    return


In [26]:
print("Patient 1")
check_dup(df_p1)
print("Patient 2")
check_dup(df_p2)

Patient 1
Number of duplicate rows: 9445

Duplicate rows:
       RESP  BP-S  BP-D  SpO2  HR  PULSE  Anomaly
6        30    81   120    98  73     73        0
7        30    81   120    98  73     73        0
9        26    81   120    98  74     74        0
13       31    81   120    98  74     75        0
17       27    81   121    98  76     75        0
...     ...   ...   ...   ...  ..    ...      ...
24959    24    78   113   100  91     91        0
24960    24    78   113   100  91     91        0
24965    24    81   117   100  94     94        0
24967    24    81   117   100  95     94        0
24975    24    82   119   100  95     96        0

[9445 rows x 7 columns]
Patient 2
Number of duplicate rows: 16551

Duplicate rows:
       RESP  BP-S  BP-D  SpO2  HR  PULSE  Anomaly
1        15    54    73    97  72     71        0
2        15    54    73    97  72     71        0
3        15    54    73    97  72     71        0
5        15    55    74    97  72     71        0
7       

- There are 9445 duplicate rows in Patient 1
- There are 16551 duplicates in Patient 2

We are not dropping the duplicates because the duplicate values could be usefull in the following analysis

### Normalization

- Algorithms like Decision Trees, Random Forests, and Gradient Boosting are not sensitive to the scale of features. These algorithms split data based on feature values rather than calculating distances or optimizing gradients, so normalization is generally not required.